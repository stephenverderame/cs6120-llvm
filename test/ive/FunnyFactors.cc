// NOLINTBEGIN
#include <cstdio>
__attribute__((optnone)) int g() { return -10; }

int main()
{
    int f = g();
    int p = g();
    for (int i = 0; i < 100; ++i) {
        int k = i * f + p;
        printf("%d\n", k);
        for (int j = 500; j >= 0; --j) {
            int m = j * k;
            printf("%d\n", m);
        }
    }

    for (int i = 100; i >= 0; --i) {
        int k = i * f + p;
        printf("%d\n", k);
    }
}
// NOLINTEND