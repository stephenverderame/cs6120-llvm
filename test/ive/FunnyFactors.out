Loop: 
22:                                               ; preds = %29, %21
  %.0 = phi i32 [ 100, %21 ], [ %30, %29 ]
  %23 = icmp sge i32 %.0, 0
  br i1 %23, label %25, label %24

Derived IV: %26 = .0 * %1
Derived IV: %27 = .0 * %1 + %2
Basic IV:   %30 = add nsw i32 %.0, -1
Basic IV:   %.0 = phi i32 [ 100, %21 ], [ %30, %29 ]
Loop: 
10:                                               ; preds = %16, %6
  %.01 = phi i32 [ 500, %6 ], [ %17, %16 ]
  %11 = icmp sge i32 %.01, 0
  br i1 %11, label %13, label %12

Derived IV: %14 = .01 * %8
Basic IV:   %17 = add nsw i32 %.01, -1
Basic IV:   %.01 = phi i32 [ 500, %6 ], [ %17, %16 ]
Loop: 
3:                                                ; preds = %23, %0
  %.02 = phi i32 [ 0, %0 ], [ %24, %23 ]
  %4 = icmp slt i32 %.02, 100
  br i1 %4, label %6, label %5

Basic IV:   %24 = add nsw i32 %.02, 1
Derived IV: %7 = .02 * %1
Derived IV: %8 = .02 * %1 + %2
Basic IV:   %.02 = phi i32 [ 0, %0 ], [ %24, %23 ]
